### PR TITLE
[CPX-Marketplace-Catalog] 2023-01-01-preview

### LINKED ISSUE #21533: [CPX-Marketplace-Catalog] API Review - 2022-09-01-preview
New API Review meeting has been requested.

**Service Name**: CPX-Marketplace-Catalog
**Review Created By**: Elad Schwartz @eladschartz 
**Review Date**: 11/30/2022 9:00 AM PT
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/21476
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: 

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=6ad796a6-5063-ed11-9562-000d3a3418aa)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by eladschartz**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [x] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

fix https://github.com/Azure/azure-rest-api-specs/issues/21533

### CODE REVIEW

**Comment by eladschartz** [comment]
Hi @BlackRider97 and @mikekistler.
I have booked a review board for this PR. Other than that is there anything else that I need to do to receive approval for this PR?

**Comment by praveenkuttappan** [comment]
Manually created swagger API review: https://apiview.dev/Assemblies/Review/43d2665540d84323b8b3be2a80dd6753

File: specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json
Comment (mikekistler) [inline specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json:34]: Please remove `x-nullable: true` here and everywhere else it appears in the file. This annotation should only be used if "null" has a special meaning that is different from the parameter/property not being set. And in this case the special meaning of "null" should be clearly described.

File: specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json
Comment (mikekistler) [inline specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json:47]: "api-version" should be a required parameter.

File: specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json
Comment (mikekistler) [inline specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json:152]: A required parameter should not specify a default value.

File: specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json
Comment (mikekistler) [inline specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json:183]: Don’t use the “$” prefix character on $select / $filter / $expand / $orderby. The [Azure API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#query-options) require query parameters to be camel case and specifically prohibit the “$” prefix on parameters borrowed from Data.

File: specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json
Comment (mikekistler) [inline specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json:201]: The orderby parameter must be all lowercase - "orderby".

**Review by mikekistler — CHANGES_REQUESTED** [review]
I flagged a number of style/structure issues.

**Comment by mikekistler** [comment]
@eladschartz Is this PR ready for re-review?

**Comment by eladschartz** [comment]
> @eladschartz Is this PR ready for re-review?

Hi @mikekistler. We still have a few decisions relating to response payload before we are ready to merge.

**Comment by eladschartz** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

Thanks for all the updates. This looks good for preview.

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (weidongxu-microsoft) [inline specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:172]: I assume "select" and "expand" would be array with collectionFormat=csv ?

Comment (weidongxu-microsoft) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:172]: Also remove the "$" from description.

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (weidongxu-microsoft) [inline specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:68]: I am not really sure the example in description is correct...

Also see Mark's comment https://github.com/Azure/azure-rest-api-specs/issues/21533#issuecomment-1332509348
that if you use "multi", prefer to call it "hideKey".

Comment (weidongxu-microsoft) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:68]: Same for the other "hideKeys" instance.

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (weidongxu-microsoft) [inline specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:131]: Is it array with csv?

**Review by weidongxu-microsoft — COMMENTED** [review]
There are many instances that I am not sure if "collectionFormat=csv" is more appropriate in Swagger, given description with example that having multiple values separated by comma.

**Comment by tjprescott** [comment]
@weidongxu-microsoft are you requesting changes or are you fine merging this?

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (eladschartz) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:131]: Storefront here can take only a single value

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (eladschartz) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:172]: Regarding select and expand, I will add that  "collectionFormat=csv " for clarity.

Comment (eladschartz) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:172]: Regarding removing the "$" from the description Will this not be confusing for users as it doesn't accurately describe the odata use case?

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (eladschartz) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:68]: Have changed both the description and the parameter name to "HideKey"

**Comment by eladschartz** [comment]
Hi @weidongxu-microsoft ,
Thank you for review. Have fixed / resolved / responded to 2 of your comments, and left an open response to the comment regarding "$" in odata params.

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (weidongxu-microsoft) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json]: @eladschartz 

Ref https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#query-options

> ⛔ DO NOT prefix any of these query parameter names with "$" (the convention in the [OData standard](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_QueryingCollections)).

**Comment by weidongxu-microsoft** [comment]
@tjprescott I think most of my comments is solved. I am waiting for @mikekistler to have one final review (since I've asked quite some changes to Swagger and @eladschartz - I couldn't find your MS alias - changed a lot).

I can merge it after Mike reviewed (will add the SignedOff label).

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (eladschartz) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json]: Ok. Have removed "$" from descriptions as well

**Comment by eladschartz** [comment]
Thank you both for your reviews
@tjprescott I think there may be a problem with the validation pipeline as tests seem to be failing unrelated to my changes. I tested this theory against another PR of mine where all I changed was the custom words file and the 3 tests that are now failing (which had passed before) failed. All failures seem related to a missing "junitCaseReport" file. Any idea how to proceed?

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
@raych1 
Looks like pipeline problem. See this in other recent PR as well. Do you know who'd able to diagnose?

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (mikekistler) [inline specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:115]: The verb in the operationId for a GET that returns a collection should be "list" rather than "get".
```suggestion
        "operationId": "PublicProducts_List",
```

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

I left a few comments most of which are easy fixes. Feel free to defer on any of these if they are more complicated. Approving to allow merge as soon as the service team has addressed these.

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (eladschartz) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:115]: Fixed

File: specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json
Comment (eladschartz) [reply specification/marketplacecatalog/data-plane/Products/preview/2023-01-01-preview/products.json:241]: fixed

**Comment by eladschartz** [comment]
Looks like pipeline issue has been fixed and all checks are passing.
@tjprescott can we please merge?

**Pull request merged**
